tech-specs/src/specs/ledger.md

```diff
@@ -0,0 +1,190 @@
+# The ledger
+
+## The protocol
+
+- TODO describe DKG transactions
+
+### Transactions
+
+A transaction [encoded with proto3](./encoding.md#transactions) received from ABCI `DeliverTx` method is executed in two main steps:
+
+1. [Transaction execution](#transaction-execution)
+1. [Validity predicates check](#validity-predicates-check)
+
+#### Transaction execution
+
+For any error encountered in any of the following steps of transaction execution, the protocol MUST charge the gas used by the transaction and discard any storage changes that the transaction attempted to perform.
```

Comment:
except for gas charged and nonce incremented (of some sort); these storage changes must always happen

Reply:
we need to increment the nonce so the transaction cannot be replayed

Reply:
I agree, but I'm not sure what would be best fitting approach for us, so I'd like to discuss a bit more about the possible ways we could add replay protection first. At a high level, we can either implement it in the ledger or in VPs. 

If it's in the ledger, then we need to decide how is the nonce/counter associated with an account as our txs don't have a single source. Maybe this could be in the DKG wrapper, which will need a source source for gas anyway, but then we also need to link this in the decrypted tx somehow.

Reply:
As discussed last week, we will include replay protection in the DKG wrapper. I left a note in the spec to describe this further.